frontend/docsettings.lua

```diff
@@ -154,6 +193,20 @@ function DocSettings:getFileFromHistory(hist_name)
     end
 end
 
+--- Returns the directory and full filepath of a hash-ID-based sidecar metadata store
+-- @string doc_path path to the document (e.g., `/foo/bar.pdf`)
+function DocSettings:getSidecarHashDirAndFilepath(doc_path)
+    -- Getting PDF ID from trailer via mupdf has not been implemented - everything uses partial MD5
+    local path = self:getSidecarDir(doc_path, "hash")
+    local filetype = doc_path:match(".+%.(%w+)$")
+    if not filetype or filetype == "" then
+        return "", ""
+    end
+    local hash_file = "metadata." .. filetype .. ".lua"
+    local hash_filepath = path .. '/' .. hash_file
```

Comment:
nit

Reply:
Also I think @hius07 mentioned it earlier but there's a lot of string creation going on there vs table.concat or string buffer, which might potentially be relevant with a few thousand files? Otherwise probably not.

Reply:
Keep in mind that LuaJIT *will* optimize some of those. I don't rightly recall all the nitty-gritty, but at the very least trivial stuff with constants like `"foo" .. "bar" .. "baz"` are done in one pass, IIRC.

TL;DR: Unless it's a hot path, I wouldn't necessarily worry about it too much.

Reply:
Is the nit the same as your second comment? I feel like this amount of string concatenation will be minimal almost always.

Reply:
Barring that, yeah, away with single quotes ;p.

Reply:
Oh yes lol. I'll definitely forget this convention but will change it.

Reply:
Resolving

Reply:
> Is the nit the same as your second comment? I feel like this amount of string concatenation will be minimal almost always.

No, the nit is the nit, sorry for adding later. Single quotes.

Reply:
@NiLuJe 
> Keep in mind that LuaJIT will optimize some of those. I don't rightly recall all the nitty-gritty, but at the very least trivial stuff with constants like "foo" .. "bar" .. "baz" is done in one pass, IIRC.

I was primarily referring to the consecutive variables but note that if you run an artificial benchmark you'll see concatenation run behind by quite a long shot due to being a suboptimal pattern. But I'm just leaving it as a comment; the author knows better if it's a relevant comment or not. Are there any plans to run it recursively for example, etc.